apis/projectcontour/v1alpha1/contourdeployment.go

```diff
@@ -151,6 +161,12 @@ type EnvoySettings struct {
 	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
 	// +optional
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// LogLevel sets the log level for Envoy
+	// Allowed values are "trace", "debug", "info", "warn", "error", "critical", "off".
+	//
+	// +optional
+	LogLevel LogLevel `json:"logLevel,omitempty"`
```

Comment:
Noting that we are reusing the same type and consts for Contour logging as for Envoy logging, but since our validation is at runtime and checks for the set of values allowed depending on which component it's for, I think it's fine.